### DISCUSSION TITLE
Production ready?

### DISCUSSION THREAD

**Discussion by softshipper** [smallrye/smallrye-graphql#1437] [Q&A]
Hi folks,
Is [smallrye-graphql](https://github.com/smallrye/smallrye-graphql) production ready or do I have to wait for V2?

Thanks

**Comment by t1** [comment]
We use it in several business-critical systems for almost 2 years in production by now. It's definitely production ready.

**Reply by softshipper** [reply]
Should I use version 2.x.x or 1.4.x?

**Reply by jmartisk** [reply]
2.x is not released yet and it's for moving to the Jakarta namespace. 
I suppose you use it either via Quarkus or WildFly, rather than directly? In which case you can't really choose which SmallRye GraphQL version you want to use, you use the one that is provided/supported by your runtime. None have moved to 2.x yet, so the latest is considered 1.6.0.

**Reply by softshipper** [reply]
I am planning to use it via Quarkus. When is the version 2.x.x ready to use?

**Reply by phillip-kruger** [reply]
Version 2 is no different from 1.4.x, as Jan said it only contains the Jakarta namespace breaking change. That will go into quarkus as soon as quarkus switch to the Jakarta namespace. For now just use the latest quarkus version and use the quarkus extension to pull in smallrye graphql

**Reply by t1** [reply]
Remember: all these versions are [semantic](https://semver.org), i.e. "the bad, the good, and the ugly", as a major update is "bad" (breaking change), a minor update is "good" (new features), and a micro update is "ugly" (bugfix).
